### ISSUE TITLE
Cannot switch the FSI language to english

### ISSUE THREAD

**Issue by blfuentes** [dotnet/fsharp#16146]
I have installed net core 7.0.402 in an OS with language settings in Spanish but without any language pack installed, so the dotnet sdk should be in english.
I have added the environment variable to force dotnet to run with english locale:
![image](https://github.com/dotnet/fsharp/assets/5121778/4d1c6d5b-6d5f-4af5-b86b-4aa0838cf4aa)
So far, this seems to work for running dotnet commands (I am leaving an error to show that the msbuild reports the errors and are displayed in english:
![image](https://github.com/dotnet/fsharp/assets/5121778/76bd1ffe-3703-4d4f-8f84-7d6517aa3748)
But whenever I execute dotnet fsi, the fsi interactive terminal switches to my OS language(Spanish)
![image](https://github.com/dotnet/fsharp/assets/5121778/0ed18d90-959d-4de3-826b-840d4fa70efe)
![image](https://github.com/dotnet/fsharp/assets/5121778/b6178b31-fe24-4d22-b3a7-ae327d2a187a)

Can I somehow change the `FSI interactive`  language to use also english?

**OS System:** Windows 10 22h2 build 19045.3570
**netcore Version** 7.0.402

Thanks!

**Comment by vzarytovskii** [comment]
I don't think we have a way, other than setting language for current console, @KevinRansom?

**Comment by baronfel** [comment]
I think the correct way to do this would be in the 'dotnet fsi' wrapper command in the dotnet/sdk repo, since this is a question of making an SDK convention apply to a tool that ships as part of the SDK. If there's a known flag/argument for controlling locale then we can set it in that wrapper before invoking the FSI assembly. This is how many integrations (Nuget, format, etc) already happen in the SDK.

**Comment by KevinRansom** [comment]
fsi has a command line option --preferreduilang to set the current language
dotnet-fsi --preferreduilang:es-ES
in your case:
dotnet-fsi --preferreduilang:en-EN

There is no persistent way of specifying this other than using the Operating System settings.

I have tested a fix that I will submit to the dotnet sdk:

````
C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>dotnet fsi

Microsoft (R) F# interactivo, versión 12.8.400.0 for F# 8.0
Copyright (C) Microsoft Corporation. Todos los derechos reservados.

Para obtener ayuda, escriba #help;;

> #quit;;

C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>set DOTNET_CLI_UI_LANGUAGE=en-En

C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>dotnet fsi

Microsoft (R) F# Interactive version 12.8.400.0 for F# 8.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> #quit;;

C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>set DOTNET_CLI_UI_LANGUAGE=

C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>dotnet fsi

Microsoft (R) F# Interactive version 12.8.400.0 for F# 8.0
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> #quit;;

C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>set DOTNET_CLI_UI_LANGUAGE=es-Es

C:\kevinransom\sdk\artifacts\bin\redist\Debug\dotnet>dotnet fsi

Microsoft (R) F# interactivo, versión 12.8.400.0 for F# 8.0
Copyright (C) Microsoft Corporation. Todos los derechos reservados.

Para obtener ayuda, escriba #help;;

>
````

**Comment by blfuentes** [comment]
thanks!

**Comment by toburger** [comment]
I am using the latest .net 8 runtime and calling `dotnet fsi` doesn't consider the `DOTNET_CLI_UI_LANGUAGE` environment variable.

Calling dotnet for the first time produces a mix of languages where a part comes from the dotnet runtime and fsi interactive:

![Image](https://github.com/user-attachments/assets/268b0e80-7d9f-43ed-acbd-b35bafc2692a)

----

OT: Funnily enough updating the workload as mentioned in the screenshot doesn't respect my language settings, again! Not that you can do anything about it from the F# side but oh my is this a mess! Why does Microsoft always want to localize all my developer tools!?

![Image](https://github.com/user-attachments/assets/3e406bdf-0bb3-4b83-8cd2-9208698fc397)

I'm not even Italian! 😆
